cli/src/cli_util.rs

```diff
@@ -1642,8 +1645,15 @@ impl WorkspaceCommandTransaction<'_> {
         formatter: &mut dyn Formatter,
         commit: &Commit,
     ) -> std::io::Result<()> {
-        // TODO: Use the disambiguation revset
-        let id_prefix_context = IdPrefixContext::new(self.helper.revset_extensions.clone());
+        // Use helper's cached IdPrefixContext if no changes made to repo
+        if !self.tx.repo().has_changes() {
+            return self.helper.write_commit_summary(formatter, commit);
```

Comment:
nit: this might be good, but appears not covered by any tests. Maybe we can go without optimized path until we find a performance problem?

Reply:
That's a good point, I hadn't noticed there weren't any existing tests covering it. The only place this seems to happen is when creating "instructions" messages for interactive editors. The main case that seems concerning to me from a performance perspective is that if somebody squashes N commits, it will unconditionally rebuild the `IdPrefixContext` N*2 times, which seems like it could add up. Maybe these should be directly calling the helper's `format_commit_summary()` instead of the transaction's?

Reply:
Yes, they can bypass tx/mut_repo if the performance really matters.

BTW, they can also bypass instruction formatting at all if not interactive. If it's interactive session, the cost of short-prefixes evaluation would be negligible anyway.

Reply:
> BTW, they can also bypass instruction formatting at all if not interactive. If it's interactive session, the cost of short-prefixes evaluation would be negligible anyway.

That sounds like a better solution to me, so I'll remove this check for now and leave that change for a future PR.